frontend/src/components/pages/HomePage/HomePage.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import {
+  HomePageStyled,
+  HomeContent,
+  DnDContent,
+  DragErrors,
+  DragWarnings
+} from './HomePage.styled';
+
+// Children
+import DragNDrop from '../../elements/DragNDrop/DragNDrop';
+import FilesList from './FilesList/FilesList';
+
+const ALLOWED_MIME_TYPES = ['application/pdf'];
+const MAX_FILES = 8;
+const MAX_FILE_SIZE = 30_000;
+
+function HomePage(props) {
+  const fileInputRef = React.createRef();
+  const [dragWarnings, setDragWarnings] = useState();
+  const [dragErrors, setDragErrors] = useState();
+  const [files, setFiles] = useState([]);
+
+  const handleDrop = drop => {
+    setDragErrors(drop.errors);
+    setDragWarnings(drop.warnings);
+    if (files.length + drop.files.length > MAX_FILES) {
+      setDragErrors(['Maximum file limit exceeded']);
+      return;
+    }
+    // TODO: Reject files with duplicate file.name
+    setFiles([...files, ...drop.files]);
+  };
+
+  const handleRemoveFile = rmFile => {
+    // browser stores a "path" to the last file uploaded on the input.
+    // It's necessary to "clear" the inputs value here, but not on drop--
+    // the browser just replaces previous files in the case of a drop.
+    if (fileInputRef.current) fileInputRef.current.value = '';
+    // TODO: It would be great to get this taken care of inside DragNDrop,
+    // TODO: but currently DragNDrop has no concept of a FilesList or removing files.
+    setFiles(files.filter(fl => fl.name !== rmFile.name));
+  };
```

Comment:
If I upload the same file multiple times, when I click to delete the duplicate(s) all files with that matching name are deleted. Only the file whose delete button has been clicked on should be deleted.

Reply:
Good call. This commit below 👇uses a happy marriage between Sets and the File object to enforce both the order of the list of files (so we can safely delete by index) and the uniqueness of files uploaded. Two birds with one stone!